### PR TITLE
[bootstrap] Allow usage of string literals for event names

### PR DESCRIPTION
**Pull request by nwalters512**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

This PR ensures that `el.addEventListener('show.bs.modal')` works the same as `el.addEventListener(window.bootstrap.Modal.Events.show)`, both of which are equally valid.

### CODE REVIEW

File: types/bootstrap/index.d.ts [hunk 1]
```diff
@@ -39,13 +39,19 @@ declare global {
 
     interface Element {
         addEventListener(
-            type: Carousel.Events,
+            type: Carousel.Events | 'slide.bs.carousel' | 'slid.bs.carousel',
```
Comment (nwalters512) [inline types/bootstrap/index.d.ts:42]: If we're willing to bump the minimum TS version to 4.1, we can take advantage of [template literal types](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#template-literal-types) to simplify this and maintain a single source of truth for event names:

```suggestion
            type: Carousel.Events | `${Carousel.Events}`,
```

I couldn't find another way to access the string values of an enum without that.

Comment (peterblazejewicz) [reply types/bootstrap/index.d.ts:42]: IMO Bootstrap is quite popular across different user bases, so if that can be implemented today, one should use TS versions:
https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-use-features-from-very-new-typescript-versions

Comment (nwalters512) [reply types/bootstrap/index.d.ts:42]: I don't particularly want to have to maintain two separate type definitions here, especially seeing as we can implement this without relying on newer language features. I'm inclined to ship this as-is using the duplicated literal types.

**Review by jakebailey — APPROVED** [review]

**Comment by nwalters512** [comment]
Ready to merge

**Pull request merged**